core/src/main/java/com/microsoft/applicationinsights/internal/quickpulse/QuickPulseNetworkHelper.java

```diff
@@ -33,7 +34,27 @@
 final class QuickPulseNetworkHelper {
     private final static long TICKS_AT_EPOCH = 621355968000000000L;
     private static final String HEADER_TRANSMISSION_TIME = "x-ms-qps-transmission-time";
-    private final static String QP_STATUS_HEADER = "x-ms-qps-subscribed";
+    private final static String QPS_STATUS_HEADER = "x-ms-qps-subscribed";
+    private final static String QPS_SERVICE_POLLING_INTERVAL_HINT  = "x-ms-qps-service-polling-interval-hint";
+    private final static String QPS_SERVICE_ENDPOINT_REDIRECT   = "x-ms-qps-service-endpoint-redirect";
+    private static final String QPS_INSTANCE_NAME_HEADER = "x-ms-qps-instance-name";
+    private static final String QPS_STREAM_ID_HEADER = "x-ms-qps-stream-id";
+    private static final String QPS_MACHINE_NAME_HEADER = "x-ms-qps-machine-name";
+    private static final String QPS_ROLE_NAME_HEADER = "x-ms-qps-role-name";
+    private static final String QPS_INVARIANT_VERSION_HEADER = "x-ms-qps-invariant-version";
+
+
+    public HttpPost buildPingRequest(Date currentDate, String address, String quickPulseId, String machineName, String roleName, String instanceName) {
+
+        HttpPost request = buildRequest(currentDate, address);
+
+        request.addHeader(QPS_ROLE_NAME_HEADER, roleName);
+        request.addHeader(QPS_MACHINE_NAME_HEADER, machineName);
+        request.addHeader(QPS_STREAM_ID_HEADER, quickPulseId);
+        request.addHeader(QPS_INSTANCE_NAME_HEADER, instanceName);
+        request.addHeader(QPS_INVARIANT_VERSION_HEADER, "2");
```

Comment:
Why 2?

Reply:
> Why 2?

checkout DefaultQuickPulsePingSender.java : 76, this is to match whats in the body

`"\"InvariantVersion\": 2," +`

Reply:
Let's bump it - here and in the body. In fact, can we please have it in a centralized location rather than in 2 separate places? We usually bump the invariant version when we add behaviors.